### ISSUE TITLE
MACE minimal NaCl example on pretrained Materials Project not working

### ISSUE THREAD

**Issue by pedrojuanbj** [ACEsuit/mace#261]
**LinAlgError when running NaCl example system provided**

There is a `LinAlgError: "singular matrix"` when running the minimal example  given for the pretrained model from Materials Project found at the https://huggingface.co/cyrusyc/mace-universal Model card. This happens for the given checkpoint and the other checkpoints in the MACE readme. 

1. Go to https://huggingface.co/cyrusyc/mace-universal and follow instructions for installation.
2. Download the pretrained model card's checkpoint or the one mentioned in the readme. 
3. Execute the code in a Jupyter Notebook for the given example. 

Run in a desktop workstation:
 - OS: Linux Ubuntu 20.04.6 LTS
 - Python 3.10.11
 - ASE: 3.22.1

**NOT WORKING EXAMPLE FROM HUGGING FACE**

```
from mace.calculators import MACECalculator
from ase import Atoms
from ase.md.npt import NPT

atoms = Atoms("NaCl")

calculator = MACECalculator(
  model_paths=/path/to/pretrained.model,
  device=device,
  default_dtype="float32" or "float64",
)

atoms.calc = calculator

npt = NPT(
  atoms=atoms,
  timestep=timestep,
  temperature_K=temperature,
  externalstress=externalstress,
)

npt.run(steps)
```

**Working example for me**

However, when using other systems built with "bulk" from ASE build, the code runs just fine, here is a minimal example of slab+ads:

- Bulk+Adsorbate Example:

```
from mace.calculators import MACECalculator
from ase import Atoms
from ase.md.npt import NPT
rom ase.build import bulk
from ase.build import add_adsorbate


slab = bulk('Fe', 'bcc', a=2.87, cubic=True)
add_adsorbate(slab, 'H', 1.5)
atoms = slab

calculator = MACECalculator(
  model_paths="path_to_model",
  device="cpu",
  default_dtype="float64",
)

atoms.calc = calculator

npt = NPT(
  atoms=atoms,
  timestep=1.0,
  temperature_K=300,
  externalstress=0,
)

npt.run(10)
```

This also works for SLAB only, by deleting the add_adsorbate part. 

HOWEVER, when using the fcc111 instead of bulk there is another error 
`NotImplementedError: Can (so far) only operate on lists of atoms where the computational box is an upper triangular matrix.`
Is there any way to fix this? Suggestions?

```
from ase.build import fcc111, add_adsorbate

slab = fcc111('Al', size=(2,2,3))
add_adsorbate(slab, 'H', 1.5, 'ontop')
slab.center(vacuum=10.0, axis=2)
atoms = slab
calculator = MACECalculator(
  model_paths="path_to_model",
  device="cpu",
  default_dtype="float64",
)

atoms.calc = calculator

npt = NPT(
  atoms=atoms,
  timestep=1.0,
  temperature_K=300,
  externalstress=0,
)

npt.run(10)
```

**Comment by gabor1** [comment]
your last error is nothing to do with MACE, it's the limitation of ASE's NPT.

**Comment by bernstei** [comment]
You can rotate your system into an orientation where the cell is upper triangular, BTW. 

I wouldn't be surprised if the first is also. Doing NPT on a system that's fundamentally non-periodic in one dimension (a slab) is ill behaved, at least without constraining certain cell directions, because there's no restoring stress. Does adding the adsorbate turn of PBC?  

Would be useful to show that it's really a MACE problem by running the same geometries (if not chemistry) with another potential.

**Comment by ilyes319** [comment]
`LinAlgError: "singular matrix"` this is usually an error of matscipy. Could you share the full error log?

**Comment by pedrojuanbj** [comment]
> your last error is nothing to do with MACE, it's the limitation of ASE's NPT.

This was an additional thing I was asking wondering if someone knew a workaround. 

> Would be useful to show that it's really a MACE problem by running the same geometries (if not chemistry) with another potential.

This is true, I just used Open Catalyst Project pretrained models and the error is the same and it comes from ASE, quick fix was to change the NPT to Langevin for example and the error is no more for neither other models or the MACE. 

> Can you try this script:
> 
> ```
> from ase import Atoms
> from ase.md.npt import NPT
> from ase.build import bulk
> from ase.build import add_adsorbate
> 
> 
> slab = bulk('Fe', 'bcc', a=2.87, cubic=True)
> add_adsorbate(slab, 'H', 1.5)
> atoms = slab
> 
> calculator = mace_mp(model="medium",device="cpu")
> 
> atoms.calc = calculator
> 
> npt = NPT(
>   atoms=atoms,
>   timestep=1.0,
>   temperature_K=300,
>   externalstress=0,
> )
> 
> npt.run(10)
> ```
> 
> This runs fine on my laptop.

As I said, the BULK and BULK+ADSORBATE examples I provided work fine for me, the problem is with the minimal example provided on the link originally, with a NaCl molecule on NPT that it spits out the singular matrix error. That is what I believe needs to be fixed first. The pull request was originally for the example on the model's card at hugging face which is pointed to by the readme on the repo. 

I would suggest to change the NaCl with ASE NPT example to ASE Langevin. I just did and now it works just fine.

Thank you for your help folks, it works now!

So, in summary:

- The provided example on Hugging Face to run the pretrained Materials Project model does not work currently
- The example has a NaCl molecule on an NPT ensemble. 
- This example now works if you change the example system to a "bulk" with/without adsorbate, code which I porvided for use in case with NPT. 
- Changing the ASE NPT to ASE Langevin would fix it for any system, all three above mentioned (NaCl, bulk, fcc111) worked fine for the example code on the pretrained models.

**Comment by pedrojuanbj** [comment]
> thanks for tracking this down.

I opened a pull request on hugging face with the fixed version with ASE Langevin.

**Comment by millet-brew** [comment]
> I opened a pull request on hugging face with the fixed version with ASE Langevin.

@pedrojuanbj could you share your pull request? I am also facing the same error and I am trying to see if it is the same problem.
